### ISSUE TITLE
After upgrade to 0.2.0: Invalid node_roles configuration: The node_roles must be provided for all elasticsearch topology elements or for none of them

### ISSUE THREAD

**Issue by calebdelnay** [elastic/terraform-provider-ec#328]
## Readiness Checklist
- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior
Terraform apply works without issue.

## Current Behavior
After upgrading to provider version `0.2.0` the below definition fails to apply due to the following error:
```
+ terraform14 apply -auto-approve -no-color tfplan

Error: failed updating deployment: 1 error occurred:

	* api error: deployments.elasticsearch.node_roles_error: Invalid node_roles configuration: The node_roles must be provided for all elasticsearch topology elements or for none of them

  on syncope-elastic.tf line 10, in resource "ec_deployment" "syncope":
  10: resource "ec_deployment" "syncope" {
```

## Terraform definition
```
locals {
  ec_region = "gcp-${local.gcp_region}"
}

data "ec_stack" "latest" {
  version_regex = "latest"
  region        = local.ec_region
}

resource "ec_deployment" "syncope" {
  name = "syncope-${local.app_env}"

  region                 = local.ec_region
  version                = data.ec_stack.latest.version
  deployment_template_id = "gcp-compute-optimized"

  elasticsearch {
    topology {
      id         = "hot_content"
      size       = "2g"
      zone_count = 1
    }
  }

  kibana {}
}
```

## Steps to Reproduce
Not 100% sure but it seems certain `ec_deployment` structures trigger the issue. We don't have `node_roles` defined so its confusing what the error is referring to. Maybe there is an implicit `node_roles` somewhere? I notice some of the examples have `apm {}` and `enterprise_search {}` but others do not. Maybe related?

## Context
We accidentally excluded the Terraform provider lock file from our repo and unexpectedly upgraded to provider version 0.2.0. Our CI build failed with the above error.

## Possible Solution
Reverting to 0.1.1 allowed us to move forward. We didn't try any changes to the `ec_deployment` resource definition.

## Your Environment
* Version used: Terraform provider? 0.2.0
* Running against Elastic Cloud SaaS or Elastic Cloud Enterprise and version: Elastic Cloud SaaS (I think) our deployment version is 7.13.1

**Comment by marclop** [comment]
Hi @calebchaney,

Thank you for reporting the bug, to clarify, the deployment was already created and on upgrading to `0.2.0` it caused the apply to error out, am I right?

Were there any changes on the `.tf` files themselves? I'd like to reproduce this bug as close as possible.

**Comment by calebdelnay** [comment]
Yes that's right. The deployment was created on provider version 0.1.1. We made no changes to any `.tf` files but because our deployment specifies a version of `data.ec_stack.latest.version` and Elasticsearch version 7.13.2 was also released recently, that triggered the plan and apply steps of our CI pipeline. We had a provider version string of `~> 0.1` and also did not have `.terraform.lock.hcl` committed either, so in the same CI run we also started using provider version `0.2.0` unintentionally. So a series of misconfigurations on our side resulted in a `terraform apply` to upgrade our deployment to version 7.13.2 using provider version 0.2.0 and it happened to fail with this bug. Hopefully that clarifies!
